functors_task_2/student/include/impl.h

```diff
@@ -62,10 +64,15 @@ class DataBrowser
     bool safeCall(const std::string &userId, T&& f) const
     {
         static_assert(std::is_same<decltype(f(nullptr)), bool>::value, "Provided Callable must return bool");
-        // find reader
-        // check for errors
-        // call functor
-        return false;
+        auto readerIterator = m_dataReaders.find(userId);
+
+        if (readerIterator == m_dataReaders.end())
+            return false;
+
+        if (readerIterator->second == nullptr)
```

Comment:
depends on style, but you can use implicit conversion to bool, it will look much simpler